docs/advanced/mapping.md

```diff
@@ -24,6 +24,32 @@ Multiple maps are supported in the latest versions by using `ledmapx.json` where
 
 Use -1 in the map for gaps/blank/null LEDs.
 
+The ledmap also allows defining the dimensions of a 2D matrix.
+In such case you do not need to set up 2D in settings.
+By specifying the `width` and `height` fields, these dimensions are used as default matrix size when the ledmap is applied.
+
+#### Mapping Requirements
+
+Mapping is an advanced feature, and for the correct results the following requirements must be fulfilled:
+
+* A same value must not appear multiple times in the mapping, except for -1.
+
+  Mapping works by reordering the  physical LEDs.
+  When an effect is rendered, the effect sets colors of a LEDs.
+  For example, a rainbow effect might set first LED of the strip as Red, the second LED as Yellow and the third LED as Blue.
+  Let's denote these physical LEDS as `LED[0]`, `LED[1]` and `LED[2]`.
+  Mapping changes these indices to `LED[map[0]]`, `LED[map[1]]` and `LED[map[2]]`, and by setting a map of `[2, 1, 0]` we can reverse the order.
+  But if the mapping contained the same value multiple times, for example `map=[0, 0, 0]`, then the effect would set the
+  colors of the LEDs `LED[map[0]] = LED[0]`, `LED[map[1]] = LED[0]` and `LED[map[2]] = LED[0]`, i.e. we would be setting the value of `LED[0]` multiple times.
```

Comment:
This would be nice, but unfortunately it does not work. Reason: ledmaps are many-to-1, so you can map 3 logical leds to the same physical led. But you cannot set the same "logical" led pixel to multiple output pixels, if this is what you wanted to say.

Your example ``"map":[0,0,0]`` does the opposite: when an effect changes pixel 0, 1 or 2, then this will always change physical led 0. So its effectively the same as ``"map":[-1,-1,0]`` because the value of pixel 2 will "win" in wled 0.16, while in wled 0.15 and earlier the result would be random.